src/ragas/metrics/_prometheus.py

```diff
@@ -0,0 +1,129 @@
+                        "score": 5,
+                    }
+                ]
+            ).dicts(),
+        }
+    ],
+    input_keys=["question", "answer", "ground_truth", "rubrics"],
+    output_key="analysis",
+    language="english",
+)
+
+class Prometheus(MetricWithLLM):
+    name = "prometheus"
+    evaluation_mode = EvaluationMode.qga  # Uses question, ground truth, answer
+
+    def __init__(
+            self,
+            mode: PrometheusMode = PrometheusMode.ABSOLUTE,
+            rubrics: Optional[Dict] = None,
+            llm: Optional[BaseRagasLLM] = None,
+            max_retries: int = 1,
+    ):
+        super().__init__(llm=llm)
+        self.mode = mode
+        self.rubrics = rubrics
+        self.max_retries = max_retries
+
+
+    async def _ascore(self, row: Dict, callbacks: t.Callbacks, is_async: bool = False) -> float:
+        if self.mode == PrometheusMode.ABSOLUTE:
+            return await self._absolute_score(row, callbacks)
+        elif self.mode == PrometheusMode.RELATIVE:
+            return await self._relative_score(row, callbacks)
+        else:
+            raise ValueError(f"Invalid mode: {self.mode}")
+
+    def _create_prompt(self, row: Dict) -> Prompt:
+        return PROMETHEUS_ABSOLUTE_PROMPT.format(
+            question=row.get('question', ''),
+            answer=row.get('answer', ''),
+            ground_truth=row.get('ground_truth', ''),
+            rubrics=self.rubrics,
+        )
+
+    async def _absolute_score(self, row: Dict, callbacks: t.Callbacks) -> float:
+        prompt_value = self._create_prompt(row)
+
+
+        response = await self.llm.generate(prompt_value, callbacks=callbacks)
+
+
+        parsed_response = await _score_feedback_output_parser.aparse(
+            response.generations[0][0].text, prompt_value, self.llm, self.max_retries
+        )
+
+        if parsed_response is None:
+            return np.nan
+
+        score = parsed_response.dicts()[0]['score']
+        return score
+
+    async def _relative_score(self, row: Dict, callbacks: t.Callbacks) -> float:
```

Comment:
the relative scoring returns feedback and relative grading. Ragas does not support this as of now as metrics output score. Ranking metric can be later added when we add support for this.